src/App.vue

```diff
@@ -1,27 +1,84 @@
 <template>
-  <img alt="Vue logo" src="./assets/logo.png" />
-  <HelloWorld msg="Welcome to Your Vue.js + TypeScript App" />
+  <section class="container">
+    <header>
+      <SearchForm v-if="!isMovieCardClicked"></SearchForm>
+      <DetailMovieInfo
+        v-if="isMovieCardClicked"
+        :movie="targetMovie"
+        @clicked="onHomePaget"
+      />
+    </header>
+    <main>
+      <MovieList
+        :movies="movies"
+        :isDetail="isMovieCardClicked"
+        :genres="genres"
+        @clicked="onDetailPage"
+      />
+    </main>
+    <FooterComponent class="footer"></FooterComponent>
+  </section>
 </template>
 
 <script lang="ts">
 import { defineComponent } from "vue";
-import HelloWorld from "./components/HelloWorld.vue";
+
+import FooterComponent from "./shared/FooterComponent.vue";
+import SearchForm from "./components/SearchForm.vue";
+import DetailMovieInfo from "./components/DetailMovieInfo.vue";
+import MovieList from "./components/MovieList.vue";
+
+import { IMovie } from "./interfaces/movie.interface";
+
+import mockedMovies from "./data/movies.json";
 
 export default defineComponent({
   name: "App",
-  components: {
-    HelloWorld,
+  components: { FooterComponent, SearchForm, MovieList, DetailMovieInfo },
+  data: function () {
+    return {
+      movies: mockedMovies.movies,
+      isMovieCardClicked: false,
+      targetMovie: {},
+      targetMovieCardIndex: 0,
+      genres: new Array<string>(),
+    };
+  },
+  methods: {
+    onDetailPage(id: string): void {
+      this.isMovieCardClicked = true;
+
+      const cloneMovies = JSON.parse(JSON.stringify(this.movies));
+
+      this.targetMovie = cloneMovies.find(
+        (movie: IMovie) => movie.id === Number(id)
+      );
+      this.targetMovieCardIndex = this.movies.findIndex(
+        (movie: IMovie) => movie.id === Number(id)
+      );
+      this.genres = this.movies[this.targetMovieCardIndex]?.genres;
+
+      this.movies = cloneMovies.filter((movie: IMovie) => {
```

Comment:
this.movies = clonedMovies.filter(({ genres }) => genres.some((genre) => this.genres.includes(genre)));